aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -198,24 +200,24 @@ void TensorIterator::compute_types() {
           }
         }
       }
+    }
+  }
```

Comment:
before type promotion was added, this was two separate loops. Combining them presumes we only need to `validate_dtype` below when `missing_dtypes || compute_common_dtype`, but that assumption was incorrect.